source/common/http/websocket/ws_handler_impl.cc

```diff
@@ -0,0 +1,145 @@
+#include "common/http/websocket/ws_handler_impl.h"
+
+#include "envoy/buffer/buffer.h"
+#include "envoy/event/dispatcher.h"
+#include "envoy/event/timer.h"
+#include "envoy/router/router.h"
+
+#include "common/buffer/buffer_impl.h"
+#include "common/common/assert.h"
+#include "common/common/enum_to_int.h"
+#include "common/common/utility.h"
+#include "common/http/codes.h"
+#include "common/http/header_map_impl.h"
+#include "common/http/headers.h"
+#include "common/http/http1/codec_impl.h"
+#include "common/http/utility.h"
+#include "common/network/utility.h"
+
+#include "spdlog/spdlog.h"
+
+namespace Envoy {
+namespace Http {
+namespace WebSocket {
+
+WsHandlerImpl::WsHandlerImpl(Http::HeaderMap& request_headers,
+                             const Router::RouteEntry* route_entry,
```

Comment:
Can it be a const reference, or should it be a shared or unique_ptr instead if it needs to be nullable?